zilencer/management/commands/render_messages.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import datetime
+from typing import Any, Generator
+
+from django.core.management.base import BaseCommand, CommandParser
+from django.db.models import QuerySet
+
+from zerver.lib.message import render_markdown
+from zerver.lib.str_utils import force_str
+from zerver.models import Message
+
+
+def queryset_iterator(queryset, chunksize=100):
+    # type: (QuerySet, int) -> Generator
+    if queryset.count():
+        queryset = queryset.order_by('pk')
+        while queryset.count():
+            for row in queryset[:chunksize]:
+                pk = row.pk
+                yield row
+            queryset = queryset.filter(pk__gt=pk)
+
+
+class Command(BaseCommand):
+    help = """
+    Render messages to a file.
+    Usage: python manage.py render_messages <destination> <--amount>
+    """
+
+    def add_arguments(self, parser):
+        # type: (CommandParser) -> None
+        parser.add_argument('destination', help='Destination folder for resulting file')
+        parser.add_argument('--amount', default=100000, required=False, help='Amount of messages')
+
+    def handle(self, *args, **options):
+        # type: (*Any, **Any) -> None
+        dest_dir = options['destination']
+        amount = options['amount']
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+        now = datetime.datetime.now()
+        with open(os.path.join(dest_dir, now.strftime('%Y-%m-%d-%H:%M:%S.txt')), 'w') as result:
+            latest = Message.objects.latest('pub_date').id
+            messages = Message.objects.filter(pk__gt=latest-amount).order_by('pk')
+            for message in queryset_iterator(messages):
+                result.write(force_str(render_markdown(message, message.content)))
```

Comment:
Let's make this format more structured, so that one can actually compare them.  Issue with this include:
* doesn't show message ID so hard to look up details
* doesn't handle newlines / record boundaries reasonably

I'd like to have a comparison tool that reads two files in the format provided by this with the same range of messages, and compares them, printing the total number of messages and the number of messages whose rendering differ between the two files.  I don't believe such a tool is possible with this format (maybe you should write it out as JSON?).  It'd be great to include the comparison tool as a second commit in this PR.

Does that make sense?  This would allow us to verify to what extent e.g. the italics changes would actually impact existing messages.